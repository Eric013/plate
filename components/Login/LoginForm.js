import { Field, reduxForm } from 'redux-form'
import { compose, graphql } from 'react-apollo'

import BorderedButton from '../BorderedButton/BorderedButton'
import Link from 'next/link'
import PropTypes from 'prop-types'
import React from 'react'
import RenderWhiteTextField from '../../utils/RenderWhiteTextField'
import cookie from 'react-cookie'
import { loginMutation } from '../../mutations/loginMutations'
import { loginValidations } from '../../validations/loginValidations'

const userLogin = async login => {
  const username = document.getElementById('username').value
  const token = await login(username)
  if (token.data.login === null) {
    alert('invalid login')
  } else {
    cookie.save('token', token.data.login, { path: '/' })
    window.location.href = '/'
  }
}

const LoginForm = ({ handleSubmit, login }) => {
  return (
    <div className="container-fluid">
      <div className="row full-height middle-xs middle-sm middle-md middle-lg center-xs center-sm center-md center-lg">
        <div className="col-xs-12 col-sm-12 col-md-6 col-lg-4">
          <h1 className="header-text">Login</h1>
          <form onSubmit={handleSubmit(() => userLogin(login))}>
            <Field
              name="username"
              id="username"
              component={RenderWhiteTextField}
              type="text"
              label="Username"
            />
            <Field
              name="password"
              id="password"
              component={RenderWhiteTextField}
              type="password"
              label="Password"
              style={{ marginBottom: 20 }}
            />
            <BorderedButton color="white" type="submit" label="Login" />
          </form>
          <div className="forgot-password">
            <Link prefetch href="/forgotpassword">
              <a>Forgot Password?</a>
            </Link>
          </div>
        </div>
      </div>
      <style jsx>
        {`
          .text-field-email {
            margin-bottom: 15px;
          }
          .text-field-password {
            margin-bottom: 15px;
          }
          .forgot-password {
            margin-top: 15px;
          }
          .forgot-password a {
            color: white;
            text-decoration: none;
          }
          .full-height {
            height: 90vh;
          }
          .header-text {
            color: white;
            margin-bottom: 50px;
          }
        `}
      </style>
    </div>
  )
}

LoginForm.propTypes = {
  handleSubmit: PropTypes.func,
  login: PropTypes.func
}

export default compose(
  reduxForm({
    form: 'loginForm',
    validate: loginValidations
  }),
  graphql(loginMutation, {
    props: ({ mutate }) => ({
      login: username => mutate({ variables: { username } })
    })
  })
)(LoginForm)
