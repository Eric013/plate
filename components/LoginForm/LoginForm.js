import { Button, Divider, Form, Grid, Icon, Message } from 'semantic-ui-react'
import { inject, observer } from 'mobx-react'

import { Component } from 'react'
import Router from 'next/router'
import { auth } from '../../lib/db'
import firebase from 'firebase'

@inject('store') @observer
export default class extends Component {

  login = (e) => {
    e.preventDefault()

    const email = document.getElementById('email').value
    const password = document.getElementById('password').value

    const promise = auth.signInWithEmailAndPassword(email, password)

    promise
      .then(() => Router.push('/dashboard'))
      .catch(() => {
        this.props.store.showLoginMessage()
      })
  }

  loginWithGoogle = () => {
    const provider = new firebase.auth.GoogleAuthProvider()

    const promise = auth.signInWithPopup(provider)

    promise
      .then(() => Router.push('/dashboard'))
  }

  loginWithFacebook = () => {
    const provider = new firebase.auth.FacebookAuthProvider()

    const promise = auth.signInWithPopup(provider)

    promise
      .then(() => Router.push('/dashboard'))
  }

  render() {
    return (
      <div>
        <Message
          floating='true'
          negative='true'
          visible={this.props.store.loginMessage}
          hidden={this.props.store.loginMessageHidden}
          onDismiss={this.props.store.hideLoginMessage}
          header='Are you sure you are registered?'
          content='We did not seem to find an account with that email'
          style={{marginTop: '40px', marginBottom: '10px'}}
        />
        <div className='login-form'>
          <Grid centered>
            <Grid.Row>
              <Grid.Column computer={8} tablet={16} mobile={16}>
                <h3>Login Into Your Account</h3>
                <Form onSubmit={this.login}>
                  <Form.Field>
                    <input type='text' name='email' id='email' placeholder='email' required />
                  </Form.Field>
                  <Form.Field>
                    <input type='password' name='password' id='password' placeholder='password' required />
                  </Form.Field>
                  <Button type='submit' primary>Login</Button>
                  <p><a href='#' className='link'>Forgot Password?</a></p>
                </Form>
              </Grid.Column>
              <Grid.Column computer={1} only='computer'>
                <Divider vertical>or</Divider>
              </Grid.Column>
              <Grid.Column computer={7} mobile={16}>
                <div className='buttons'>
                  <Button color='facebook' onClick={this.loginWithFacebook} style={{marginBottom: '15px', width: '300px'}}>
                    <Icon name='facebook' /> Sign in with Facebook
                  </Button>
                  <br />
                  <Button color='google plus' onClick={this.loginWithGoogle} style={{marginBottom: '15px', width: '300px'}}>
                    <Icon name='google plus' /> Sign in with Google Plus
                  </Button>
                  <br />
                  <Button color='twitter' style={{width: '300px'}}>
                    <Icon name='twitter' /> Sign in with Twitter
                  </Button>
                </div>
              </Grid.Column>
            </Grid.Row>
          </Grid>
          <style jsx>{`
            .login-form {
              background-color: #F7F7F7;
              border-radius: 10px;
              padding: 50px;
              margin-top: 50px;
              width: 100%;
              display: flex;
              justify-content: center;
            }
            .link {
              margin-top: 10px;
              padding-top: 10px;
            }
            .buttons {
              margin-top: 50px;
            }
          `}</style>
        </div>
      </div>
    )
  }
}
