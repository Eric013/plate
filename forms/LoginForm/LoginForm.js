import { Field, reduxForm } from "redux-form";

import BorderedButton from "../../components/BorderedButton/BorderedButton";
import Link from "next/link";
import React from "react";
import TextField from "material-ui/TextField";

const login = () => {
  console.log("submitted");
};

type Props = {
  handleSubmit?: Function
};

const LoginForm = ({ handleSubmit }: Props) => (
  <div className="container-fluid">
    <div
      className="row full-height middle-xs middle-sm middle-md middle-lg middle-xl center-xs center-sm center-md center-lg center-lg"
    >
      <div className="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4">
        <h1 className="header-text">Login</h1>
        <form onSubmit={handleSubmit(login)}>
          <div className="text-field-email">
            <TextField
              hintStyle={{ color: "white" }}
              inputStyle={{ color: "white" }}
              autoComplete={"off"}
              id="email"
              fullWidth
              required
            >
              <Field
                name="email"
                component="input"
                type="email"
                placeholder="Email"
              />
            </TextField>
          </div>
          <div className="text-field-password">
            <TextField
              autoComplete={"off"}
              inputStyle={{ color: "white" }}
              hintStyle={{ color: "white" }}
              id="password"
              fullWidth
              required
            >
              <Field
                name="password"
                component="input"
                type="password"
                placeholder="Password"
              />
            </TextField>
          </div>
          <BorderedButton type="submit" label="Login" />
        </form>
        <div className="forgot-password">
          <Link prefetch href="/forgotPassword">
            <a>Forgot Password?</a>
          </Link>
        </div>
      </div>
    </div>
    <style jsx>
      {
        `
          .text-field-email {
            margin-bottom: 15px;
          }
          .text-field-password {
            margin-bottom: 15px;
          }
          .forgot-password {
            margin-top: 15px;
          }
          .forgot-password a {
            color: white;
            text-decoration: none;
          }
          .full-height {
            height: 90vh;
          }
          .header-text {
            color: white;
            margin-bottom: 50px;
          }
        `
      }
    </style>
  </div>
);

export default reduxForm({
  form: "loginForm"
})(LoginForm);
