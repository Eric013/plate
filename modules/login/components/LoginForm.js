import { Button, Form, Input } from 'semantic-ui-react'
import React, { PropTypes } from 'react'

import ErrorMessage from '../../../components/errorMessage/ErrorMessage'

const propTypes = {
  userLogin: PropTypes.func,
  errorState: PropTypes.string,
  errorMessage: PropTypes.string,
  loading: PropTypes.bool
}

function LoginForm({ userLogin, errorState, errorMessage, loading }) {
  return (
    <Form method='post' onSubmit={userLogin} size='large'>
      <ErrorMessage
        open={errorState}
        message={errorMessage}
      />
      <br />
      <Form.Field>
        <Input
          required
          fluid
          icon='user'
          iconPosition='left'
          name='email'
          placeholder='E-mail address'
          type='email'
        />
      </Form.Field>
      <Form.Field>
        <Input
          required
          fluid
          icon='lock'
          iconPosition='left'
          name='password'
          placeholder='Password'
          type='password'
        />
      </Form.Field>
      <Button type='submit' size='large' color='grey' loading={loading} fluid>LOGIN</Button>
    </Form>
  )
}

LoginForm.propTypes = propTypes

export default LoginForm
