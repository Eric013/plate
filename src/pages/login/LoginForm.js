import { Button, Form } from 'semantic-ui-react'
import React, { PropTypes } from 'react'

import ErrorMessage from '../../components/errorMessage/ErrorMessage'

const propTypes = {
  userLogin: PropTypes.func,
  errorState: PropTypes.string,
  errorMessage: PropTypes.string,
  loading: PropTypes.bool
}

function LoginForm({ userLogin, errorState, errorMessage, loading }) {
  return (
    <Form method='post' onSubmit={userLogin}>
      <ErrorMessage
        open={errorState}
        message={errorMessage}
      />
      <br />
      <Form.Field>
        <input
          id='email'
          name='email'
          placeholder='Please enter your email'
          required
          type='email'
        />
      </Form.Field>
      <Form.Field>
        <input
          id='password'
          name='password'
          placeholder='Please enter your password'
          required
          type='password'
        />
      </Form.Field>
      <input type="hidden" name="csrf_token" value="******" />
      <Button type='submit' loading={loading} color='pink' fluid>Login</Button>
    </Form>
  )
}

LoginForm.propTypes = propTypes

export default LoginForm
