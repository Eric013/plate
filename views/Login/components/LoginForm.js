import { compose, graphql } from 'react-apollo'
import { injectState, provideState } from 'freactal'

import Alert from '../../../components/Alert/Alert'
import Link from 'next/link'
import LoginMutation from '../../../mutations/login'
import PropTypes from 'prop-types'
import React from 'react'
import Router from 'next/router'
import { saveAccessToken } from '../../../utils/cookieUtils'

const wrapComponentWithState = provideState({
  initialState: () => ({
    loginError: false
  }),
  effects: {
    showError: () => state => Object.assign({}, state, { loginError: true })
  }
})

const userLogin = async (e, login, showError) => {
  e.preventDefault()
  const username = document.getElementById('username').value
  const password = document.getElementById('password').value

  const token = await login(username, password)
  if (token.data.login === null) {
    showError()
  } else {
    saveAccessToken(token.data.login, { path: '/', expires: 7 })
    Router.push('/dashboard')
  }
}

const LoginForm = wrapComponentWithState(
  injectState(({ state, effects, login }) => {
    return (
      <form onSubmit={e => userLogin(e, login, effects.showError)}>
        <fieldset>
          <legend>Login</legend>
          <div className="input-group fluid">
            <label className="input-label" htmlFor="username">
              Username:
            </label>
            <input type="text" id="username" placeholder="username" required />
          </div>
          <div className="input-group fluid">
            <label className="input-label" htmlFor="pwd">Password:</label>
            <input
              type="password"
              id="password"
              placeholder="password"
              required
            />
          </div>
          <div className="input-group fluid">
            <button type="submit" className="primary">Login</button>
            <button type="button">
              <Link href="/forgotpassword" prefetch>
                <a>Forgot password?</a>
              </Link>
            </button>
          </div>
          <div className="register-link">
            <Link prefetch href="/register">
              <a>Not Already A Member?</a>
            </Link>
          </div>
          {state.loginError && <Alert message="Error! Invalid login" />}
        </fieldset>
        <style jsx>{`
          .input-label {
            width: 80px;
          }
          a {
            text-decoration: none;
          }
          .error-text {
            text-align: center;
          }
          .register-link {
            text-align: center;
          }
        `}</style>
      </form>
    )
  })
)

LoginForm.propTypes = {
  login: PropTypes.func
}

export default compose(
  graphql(LoginMutation, {
    props: ({ mutate }) => ({
      login: (username, password) =>
        mutate({ variables: { username, password } })
    })
  })
)(LoginForm)
