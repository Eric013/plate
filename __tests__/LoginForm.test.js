/* global it, expect, describe */

import LoginForm from '../components/LoginForm/LoginForm'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<LoginForm />, div)
})

describe('Login snapshot', () => {
  it('renders without crashing!"', () => {
    const component = renderer.create(<LoginForm />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
