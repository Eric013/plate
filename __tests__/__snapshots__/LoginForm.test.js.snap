exports[`LoginForm snapshot renders without crashing!" 1`] = `
<div
  data-jsx={4290012279}>
  <div
    className="nav-bar"
    data-jsx={2548831492}>
    <div
      className="ui pointing top fixed menu">
      <a
        data-jsx={2548831492}
        href="/"
        onClick={[Function]}>
        <div
          className="link item"
          onClick={[Function]}>
          <img
            data-jsx={2548831492}
            src="/static/plateLogo.png" />
          <span
            className="logo-text"
            data-jsx={2548831492}>
            late
          </span>
        </div>
      </a>
      <div
        className="right menu">
        <div
          className="link item"
          onClick={[Function]}>
          <span
            data-jsx={2548831492}>
            MENU
          </span>
        </div>
      </div>
    </div>
    <div
      data-jsx={2779333434} />
  </div>
  <div
    className="ui centered padded grid">
    <div
      className="row">
      <div
        className="six wide computer six wide large screen sixteen wide mobile ten wide tablet column">
        <h2>
          Login
        </h2>
        <form
          className="ui form"
          onSubmit={[Function]}>
          <div
            className="field">
            <label>
              Email
            </label>
            <div
              className="ui input">
              <input
                name="email"
                placeholder="email"
                required={undefined}
                type="text" />
            </div>
          </div>
          <div
            className="field">
            <label>
              Password
            </label>
            <div
              className="ui input">
              <input
                name="password"
                placeholder="password"
                required={undefined}
                type="password" />
            </div>
          </div>
          <button
            className="ui fluid primary button"
            onClick={[Function]}
            tabIndex={undefined}>
            Login
          </button>
        </form>
      </div>
    </div>
    <a
      href="/forgotPassword"
      onClick={[Function]}>
      Forgot Password?
    </a>
  </div>
</div>
`;
